src/features/user/MFV2/ForestProgress/microComponents/ForestProgressItem.tsx

```diff
@@ -0,0 +1,312 @@
+import {
+  EditTargetIcon,
+  TreesPlantedIcon,
+  AreaRestoredIcon,
+  ConservedAreaIcon,
+} from '../../../../../../public/assets/images/icons/ProgressBarIcons';
+import ProgressBarStyle from '../ForestProgress.module.scss';
+import themeProperties from '../../../../../theme/themeProperties';
+import { useTranslations } from 'next-intl';
+import { getAchievedTarget } from '../../../../../utils/myForestV2Utils';
+import { useMemo } from 'react';
+import { useRouter } from 'next/router';
+
+type DataType = 'treesPlanted' | 'areaRestored' | 'areaConserved';
+
+type EditButtonProp = Omit<
+  ForestProgressItemProps,
+  'gift' | 'personal' | 'checked'
+>;
+export interface ProgressDataProps {
+  giftPercentage: number;
+  personalPercentage: number;
+  gift: number;
+  personal: number;
+  dataType: DataType;
+  target: number;
+  checked: boolean;
+}
+interface ForestProgressItemProps {
+  handleOpen: () => void;
+  dataType: DataType;
+  target: number;
+  gift: number;
+  personal: number;
+  checked: boolean;
+}
+
+const EditButton = ({ handleOpen, target, dataType }: EditButtonProp) => {
+  const tProfile = useTranslations('Profile');
+  const { primaryDarkColor, electricPurpleColor, mediumBlueColor } =
+    themeProperties;
+  const getDarkerColor = useMemo(() => {
+    switch (dataType) {
+      case 'treesPlanted':
+        return primaryDarkColor;
+      case 'areaRestored':
+        return electricPurpleColor;
+      case 'areaConserved':
+        return mediumBlueColor;
+      default:
+        return '';
+    }
+  }, [dataType]);
```

Comment:
I feel like using css for these styles would make things clearer. You could simply provide `dataType` as a class name.